recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,260 @@
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if (self.settings.compiler == "gcc"
+                    and tools.Version(self.settings.compiler.version) < "6.0"):
+                raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+                See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+
+    def requirements(self):
+        self.requires("aws-c-common/0.4.25")
+        self.requires("aws-c-event-stream/0.1.5")
+        self.requires("zlib/1.2.11")
+        if self.settings.os != "Windows":
+            self.requires("libcurl/7.74.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        # Keeping these unused folders make the source copy fail on Windows
+        tools.rmdir(os.path.join(self._source_subfolder, "code-generation", "generator"))
+        tools.rmdir(os.path.join(self._source_subfolder, "aws-cpp-sdk-core-tests"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        build_only = ["core"]
+        for sdk in self.sdks:
+            if getattr(self.options, sdk):
+                build_only.append(sdk)
+        self._cmake.definitions["BUILD_ONLY"] = ";".join(build_only)
+
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["MINIMIZE_SIZE"] = self.options.min_size
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in self.settings.compiler.runtime
+
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.components["core"].libs = ["aws-cpp-sdk-core"]
+        self.cpp_info.components["core"].requires = [
+                "aws-c-common::aws-c-common-lib",
+                "aws-c-event-stream::aws-c-event-stream-lib",
+                "zlib::zlib",
```

Comment:
Are `aws-c-common` and `zlib` direct dependencies?

Content of `aws-cpp-sdk-core-targets.cmake` with Visual Studio:

```cmake
# Create imported target aws-cpp-sdk-core
add_library(aws-cpp-sdk-core STATIC IMPORTED)

set_target_properties(aws-cpp-sdk-core PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "AWS_SDK_VERSION_MAJOR=1;AWS_SDK_VERSION_MINOR=8;AWS_SDK_VERSION_PATCH=130;USE_WINDOWS_DLL_SEMANTICS"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:Userenv>;\$<LINK_ONLY:version>;\$<LINK_ONLY:ws2_32>;\$<LINK_ONLY:Wininet>;\$<LINK_ONLY:winhttp>;\$<LINK_ONLY:Bcrypt>;AWS::aws-c-event-stream"
)
```

No `AWS::aws-c-common` nor `ZLIB::ZLIB` targets in the dependencies